src/main/kotlin/tech/thatgravyboat/skyblockpv/utils/CodecUtils.kt

```diff
@@ -22,4 +22,22 @@ object CodecUtils {
         { TagParser.QUICK_TEXT_SAFE.parseText(it, ParserContext.of()) },
         { it.string },
     )
+
+    val CUMULATIVE_STRING_INT_MAP: Codec<List<Map<String, Int>>> = Codec.unboundedMap(Codec.STRING, Codec.INT).listOf()
+        .xmap(
+            {
+                it.runningFold(
+                    mutableMapOf(),
+                ) { acc: MutableMap<String, Int>, mutableMap: MutableMap<String, Int>? ->
+                    LinkedHashMap(
```

Comment:
this is gonna create n number of linked hash maps per element should just use a normal for loop